lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,699 @@
+        lockTimeout = deal.makerCltvDelta!;
+        secret = deal.rPreimage!;
+        const executeTransfer = this.executeHashLockTransfer({
+          assetId: tokenAddress,
+          amount: `${amount}`,
+          timelock: lockTimeout.toString(),
+          lockHash: `0x${deal.rHash}`,
+          recipient: deal.destination!,
+        });
+        await executeTransfer;
+      }
+      return secret;
+    } catch (err) {
+      switch (err.code) {
+        case 'ECONNRESET':
+        case errorCodes.UNEXPECTED:
+        case errorCodes.TIMEOUT:
+        case errorCodes.SERVER_ERROR:
+        case errorCodes.INVALID_TOKEN_PAYMENT_RESPONSE:
+          throw swapErrors.UNKNOWN_PAYMENT_ERROR(err.message);
+        default:
+          throw swapErrors.FINAL_PAYMENT_ERROR(err.message);
+      }
+    }
+  }
+
+  public addInvoice = async (
+    { rHash: expectedHash, units: expectedUnits, expiry: expectedTimelock, currency: expectedCurrency }:
+    { rHash: string, units: number, expiry?: number, currency?: string },
+  ) => {
+    if (!expectedCurrency) {
+      throw new Error('cannot add invoice, expected currency is missing');
+    }
+    if (!expectedTimelock) {
+      throw new Error('cannot add invoice, expiry is missing');
+    }
+    const expectedTokenAddress = this.getTokenAddress(expectedCurrency);
+    const expectedIncomingTransfer: ExpectedIncomingTransfer = {
+      rHash: expectedHash,
+      units: expectedUnits,
+      expiry: expectedTimelock,
+      tokenAddress: expectedTokenAddress,
+    };
+    // TODO: should we also store this in the database?
+    this.expectedIncomingTransfers.set(expectedHash, expectedIncomingTransfer);
+  }
+
+  /**
+   * Resolve a HashLock Transfer on the Connext network.
+   */
+  public settleInvoice = async (_rHash: string, rPreimage: string) => {
+    await this.sendRequest('/hashlock-resolve', 'POST', {
+      lockHash: `0x${rPreimage}`,
+    });
+  }
+
+  public removeInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  private async getHashLockStatus(lockHash: string) {
+    const res = await this.sendRequest(`/hashlock-status/${lockHash}`, 'GET');
+    const { status } = await parseResponseBody<ConnextTransferStatus>(res);
+    return status;
+  }
+
+  public lookupPayment = async (rHash: string) => {
+    try {
+      const status = await this.getHashLockStatus(rHash);
+
+      switch (status) {
+        case 'PENDING':
+          return { state: PaymentState.Pending };
+        case 'COMPLETED':
+          return { state: PaymentState.Succeeded };
+        case 'EXPIRED':
+        case 'FAILED':
+          return { state: PaymentState.Failed };
+        default:
+          return { state: PaymentState.Failed };
+      }
+    } catch (e) {
+      throw errors.PAYMENT_NOT_FOUND;
+    }
+  }
+
+  public getRoute = async () => {
+    return {
+      // TODO: verify that it's OK to return
+      // static getTotalTimeLock
+      getTotalTimeLock: () => 101,
```

Comment:
Does 101 have any significance for Connext?

Reply:
Not really. 101 should be good enough for Connext. I've updated the comment above (not sure why it's not showing up on github).